src/components/TodoInfo.tsx

```diff
@@ -0,0 +1,116 @@
+import React, {
+  ChangeEvent, FormEvent, useEffect, useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo;
+  deletedTodosId: number[] | [];
+  handleDeletedTodo: (id: number) => void;
+  handleUpdatedTodo: (ids: number[], value: Partial<Todo>) => void;
+};
+
+export const TodoInfo: React.FC<Props> = ({
+  todo,
+  deletedTodosId,
+  handleDeletedTodo,
+  handleUpdatedTodo,
+}) => {
+  const { title, completed, id } = todo;
+  const isDeleted = deletedTodosId.some(todoId => todoId === id);
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const handleDeleteButton = (todoId: number) => {
+    if (handleDeletedTodo) {
+      handleDeletedTodo(todoId);
+    }
+  };
+
+  const handleTitleEdit = (
+    event:
+    FormEvent<HTMLFormElement>
+    | ChangeEvent<HTMLInputElement>,
+  ) => {
+    event.preventDefault();
+
+    if (newTitle === '') {
+      handleDeleteButton(id);
+
+      return;
+    }
+
+    if (handleUpdatedTodo && title !== newTitle) {
+      handleUpdatedTodo([id], { title: newTitle });
+    } else {
+      setIsEditing(false);
+    }
+  };
+
+  const handleChangeStatus = () => {
+    if (handleUpdatedTodo) {
+      handleUpdatedTodo([id], { completed: !completed });
+    }
+  };
+
+  useEffect(() => {
+    setIsEditing(false);
+  }, [todo]);
+
+  return (
+    <div className={classNames('todo', { completed })}>
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          onClick={() => handleChangeStatus}
+        />
+      </label>
+
+      {isEditing
+        ? (
+          <form onSubmit={handleTitleEdit}>
+            <input
+              type="text"
+              className="todo__title-field"
+              value={newTitle}
+              onChange={(event) => setNewTitle(event.target.value)}
+              onBlur={(event) => handleTitleEdit(event)}
+              onKeyUp={(event) => (
+                event.key === 'Escape' && handleTitleEdit(event)
+              )}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)